dibe/views.py

```diff
@@ -0,0 +1,66 @@
+from django.shortcuts import render
+
+# Create your views here.
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.status import HTTP_401_UNAUTHORIZED
+from rest_framework.authtoken.models import Token
+from django.contrib.auth import authenticate
+from .models import User
+from .serializers import UserSerializer
+from django.http import Http404
+
+class UserList(APIView):
+    def get(sefl, request):
+        users = User.objects.all()
+        serializer = UserSerializer(users, many = True)
+        return Response(serializer.data)
+
+    def post(self, request, format=None):
+        serializer = UserSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+class UserDetail(APIView):
+    """
+    Retrieve, update or delete a user instance.
+    """
+    def get_object(self, pk):
```

Comment:
get_object này là private method của cái class UserDetail (sẽ không được xài ở ngoài UserDetail) thì a nên để thêm underscored ở trước nữa để khỏi bị nhầm (_get_object(self, pk))
https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables